Code/Framework/AzCore/AzCore/Component/EntitySerializer.cpp

```diff
@@ -207,4 +209,36 @@ namespace AZ
             }
         }
     }
+
+    void DeprecatedComponentMetadata::SetShouldTrackDeprecatedCallback(ShouldTrackDeprecatedCallback callback)
+    {
+        m_shouldTrackDeprecatedCallback = callback;
+    }
+
+    void DeprecatedComponentMetadata::AddComponent(const TypeId& componentType)
+    {
+        if (!m_shouldTrackDeprecatedCallback || m_shouldTrackDeprecatedCallback())
```

Comment:
If the callback is not provided, then why do we need to add the deprecated component types? Seems like this should be just if (m_shouldTrackDeprecatedCallback()) ?

Reply:
If a callback is not provided, then it assumes it always wants to track. The !m_shouldTrackDeprecatedCallback is also there as a null check.